File: x10-2.4.3-src/x10.dist/bin/simpleLoop.java

@x10.runtime.impl.java.X10Generated
public class simpleLoop extends x10.core.Ref implements x10.serialization.X10JavaSerializable
{
    public static final x10.rtt.RuntimeType<simpleLoop> $RTT = 
        x10.rtt.NamedType.<simpleLoop> make("simpleLoop",
                                            simpleLoop.class);
    
    public x10.rtt.RuntimeType<?> $getRTT() { return $RTT; }
    
    public x10.rtt.Type<?> $getParam(int i) { return null; }
    
    public static x10.serialization.X10JavaSerializable $_deserialize_body(simpleLoop $_obj, x10.serialization.X10JavaDeserializer $deserializer) throws java.io.IOException {
        if (x10.runtime.impl.java.Runtime.TRACE_SER) { x10.runtime.impl.java.Runtime.printTraceMessage("X10JavaSerializable: $_deserialize_body() of " + simpleLoop.class + " calling"); } 
        return $_obj;
    }
    
    public static x10.serialization.X10JavaSerializable $_deserializer(x10.serialization.X10JavaDeserializer $deserializer) throws java.io.IOException {
        simpleLoop $_obj = new simpleLoop((java.lang.System[]) null);
        $deserializer.record_reference($_obj);
        return $_deserialize_body($_obj, $deserializer);
    }
    
    public void $_serialize(x10.serialization.X10JavaSerializer $serializer) throws java.io.IOException {
        
    }
    
    // constructor just for allocation
    public simpleLoop(final java.lang.System[] $dummy) {
        
    }
    
    

    
    
    //#line 7 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
    public static class $Main extends x10.runtime.impl.java.Runtime
    {
        // java main method
        public static void main(java.lang.String[] args) {
            // start native runtime
            new $Main().start(args);
        }
        
        // called by native runtime inside main x10 thread
        public void runtimeCallback(final x10.core.Rail<java.lang.String> args) {
            // call the original app-main method
            simpleLoop.main(args);
        }
    }
    
    // the original app-main method
    public static void main(final x10.core.Rail<java.lang.String> args) {
        final int nthreads$246 = 2;
        final int ksize$247 = 2;
        int i$516 = 0;
        {
            final java.lang.String[] args$value$539 = ((java.lang.String[])args.value);
            for (;
                 true;
                 ) {
                final int t$517 = i$516;
                final long t$518 = ((long)(((int)(t$517))));
                final long t$519 = ((x10.core.Rail<java.lang.String>)args).size;
                final boolean t$520 = ((t$518) < (((long)(t$519))));
                if (!(t$520)) {
                    break;
                }
                final java.lang.String t$497 = ((java.lang.String)args$value$539[(int)0L]);
                x10.runtime.impl.java.EvalUtils.eval(java.lang.Long.parseLong(t$497));
                final int t$498 = i$516;
                final int t$499 = ((t$498) + (((int)(1))));
                i$516 = t$499;
            }
        }
        long chunkStart$521 = 0;
        long chunkEnd$522 = 10L;
        final int nthreads$523 = 4;
        final long nIterations$524 = 10L;
        
        //#line 11 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
        long i$525 = 0L;
        final long chunkSize$526 = 1;
        {
            
            //#line 11 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
            x10.lang.Runtime.ensureNotInAtomic();
            
            //#line 11 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
            final x10.lang.FinishState x10$__var0 = x10.lang.Runtime.startFinish();
            {
                
                //#line 11 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                final long[] $chunkStart$521$537 = new long[1];
                
                //#line 11 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                $chunkStart$521$537[(int)0]=chunkStart$521;
                
                //#line 11 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                final long[] $chunkEnd$522$538 = new long[1];
                
                //#line 11 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                $chunkEnd$522$538[(int)0]=chunkEnd$522;
                
                //#line 11 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                try {{
                    {
                        long ii$527 = 0;
                        for (;
                             true;
                             ) {
                            final long t$528 = ii$527;
                            final long t$529 = ((long)(((int)(nthreads$523))));
                            final boolean t$530 = ((t$528) < (((long)(t$529))));
                            if (!(t$530)) {
                                break;
                            }
                            final long ni$508 = ii$527;
                            x10.lang.Runtime.runAsync(((x10.core.fun.VoidFun_0_0)(new simpleLoop.$Closure$0($chunkStart$521$537, $chunkEnd$522$538))));
                            final long t$513 = ii$527;
                            final long t$514 = ((long)(((int)(1))));
                            final long t$515 = ((t$513) + (((long)(t$514))));
                            ii$527 = t$515;
                        }
                    }
                }}catch (java.lang.Throwable __lowerer__var__0__) {
                    
                    //#line 11 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                    x10.lang.Runtime.pushException(((java.lang.Throwable)(__lowerer__var__0__)));
                    
                    //#line 11 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                    throw new java.lang.RuntimeException();
                }finally {{
                     
                     //#line 11 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                     x10.lang.Runtime.stopFinish(((x10.lang.FinishState)(x10$__var0)));
                 }}
                
                //#line 11 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                chunkStart$521 = ((long)$chunkStart$521$537[(int)0]);
                
                //#line 11 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                chunkEnd$522 = ((long)$chunkEnd$522$538[(int)0]);
                }
            }
        
        //#line 25 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
        final x10.io.Printer t$495 = ((x10.io.Printer)(x10.io.Console.get$OUT()));
        long t$490 =  0;
        if (true) {
            t$490 = 10L;
        } else {
            t$490 = 0L;
        }
        final long t$491 = t$490;
        final long t$492 = 10L;
        final long t$493 = 200L;
        final long t$494 = 200L;
        final long t$496 = 200L;
        
        //#line 25 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
        t$495.println(x10.core.Long.$box(t$496));
        }
    
    
    //#line 6 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
    final public simpleLoop simpleLoop$$this$simpleLoop() {
        
        //#line 6 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
        return simpleLoop.this;
    }
    
    
    //#line 6 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
    // creation method for java code (1-phase java constructor)
    public simpleLoop() {
        this((java.lang.System[]) null);
        simpleLoop$$init$S();
    }
    
    // constructor for non-virtual call
    final public simpleLoop simpleLoop$$init$S() {
         {
            
            //#line 6 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
            
            
            //#line 6 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
            this.__fieldInitializers_simpleLoop();
        }
        return this;
    }
    
    
    
    //#line 6 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
    final public void __fieldInitializers_simpleLoop() {
        
    }
    
    @x10.runtime.impl.java.X10Generated
    final public static class $Closure$0 extends x10.core.Ref implements x10.core.fun.VoidFun_0_0, x10.serialization.X10JavaSerializable
    {
        public static final x10.rtt.RuntimeType<$Closure$0> $RTT = 
            x10.rtt.StaticVoidFunType.<$Closure$0> make($Closure$0.class,
                                                        new x10.rtt.Type[] {
                                                            x10.core.fun.VoidFun_0_0.$RTT
                                                        });
        
        public x10.rtt.RuntimeType<?> $getRTT() { return $RTT; }
        
        public x10.rtt.Type<?> $getParam(int i) { return null; }
        
        public static x10.serialization.X10JavaSerializable $_deserialize_body(simpleLoop.$Closure$0 $_obj, x10.serialization.X10JavaDeserializer $deserializer) throws java.io.IOException {
            if (x10.runtime.impl.java.Runtime.TRACE_SER) { x10.runtime.impl.java.Runtime.printTraceMessage("X10JavaSerializable: $_deserialize_body() of " + $Closure$0.class + " calling"); } 
            $_obj.$chunkStart$521$537 = $deserializer.readObject();
            $_obj.$chunkEnd$522$538 = $deserializer.readObject();
            return $_obj;
        }
        
        public static x10.serialization.X10JavaSerializable $_deserializer(x10.serialization.X10JavaDeserializer $deserializer) throws java.io.IOException {
            simpleLoop.$Closure$0 $_obj = new simpleLoop.$Closure$0((java.lang.System[]) null);
            $deserializer.record_reference($_obj);
            return $_deserialize_body($_obj, $deserializer);
        }
        
        public void $_serialize(x10.serialization.X10JavaSerializer $serializer) throws java.io.IOException {
            $serializer.write(this.$chunkStart$521$537);
            $serializer.write(this.$chunkEnd$522$538);
            
        }
        
        // constructor just for allocation
        public $Closure$0(final java.lang.System[] $dummy) {
            
        }
        
        
    
        
        public void $apply() {
            try {{
                long kk$509 = ((long)this.$chunkStart$521$537[(int)0]);
                for (;
                     true;
                     ) {
                    final long t$510 = kk$509;
                    final long t$511 = ((long)this.$chunkEnd$522$538[(int)0]);
                    final boolean t$512 = ((t$510) < (((long)(t$511))));
                    if (!(t$512)) {
                        break;
                    }
                    
                    //#line 16 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                    long j$502 = 0L;
                    
                    //#line 16 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                    for (;
                         true;
                         ) {
                        
                        //#line 16 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                        final long t$503 = j$502;
                        
                        //#line 16 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                        final boolean t$504 = ((t$503) < (((long)(20L))));
                        
                        //#line 16 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                        if (!(t$504)) {
                            
                            //#line 16 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                            break;
                        }
                        
                        //#line 18 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                        if (true) {
                            
                        }
                        
                        //#line 16 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                        final long t$500 = j$502;
                        
                        //#line 16 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                        final long t$501 = ((t$500) + (((long)(1L))));
                        
                        //#line 16 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                        j$502 = t$501;
                    }
                    final long t$505 = kk$509;
                    final long t$506 = ((long)(((int)(1))));
                    final long t$507 = ((t$505) + (((long)(t$506))));
                    kk$509 = t$507;
                }
            }}catch (java.lang.Error __lowerer__var__0__) {
                throw __lowerer__var__0__;
            }catch (java.lang.Throwable __lowerer__var__1__) {
                throw x10.rtt.Types.EXCEPTION.isInstance(__lowerer__var__1__) ? (java.lang.RuntimeException)(__lowerer__var__1__) : new x10.lang.WrappedThrowable(__lowerer__var__1__);
            }
        }
        
        public long[] $chunkStart$521$537;
        public long[] $chunkEnd$522$538;
        
        public $Closure$0(final long[] $chunkStart$521$537, final long[] $chunkEnd$522$538) {
             {
                this.$chunkStart$521$537 = $chunkStart$521$537;
                this.$chunkEnd$522$538 = $chunkEnd$522$538;
            }
        }
        
    }
    
    }
    
    

File: x10-2.4.3-src/x10.dist/bin/simpleLoop.cc
/*************************************************/
/* START of simpleLoop */
#include <simpleLoop.h>

#include <x10/lang/Rail.h>
#include <x10/lang/String.h>
#include <x10/io/Printer.h>
#include <x10/io/Console.h>
#include <x10/lang/Any.h>
#include <x10/lang/Runtime.h>
#include <x10/lang/FinishState.h>
#include <x10/lang/CheckedThrowable.h>
#include <x10/lang/Long.h>
#include <x10/lang/Boolean.h>
#include <x10/lang/VoidFun_0_0.h>
#include <chunk.h>
#include <x10/lang/Error.h>
#include <x10/lang/Exception.h>
#include <x10/compiler/AsyncClosure.h>
#include <x10/compiler/Finalization.h>
#include <x10/compiler/Abort.h>
#include <x10/compiler/CompilerFlags.h>
#include <x10/compiler/Synthetic.h>
#ifndef SIMPLELOOP__CLOSURE__1_CLOSURE
#define SIMPLELOOP__CLOSURE__1_CLOSURE
#include <x10/lang/Closure.h>
#include <x10/lang/VoidFun_0_0.h>
class simpleLoop__closure__1 : public ::x10::lang::Closure {
    public:
    
    static ::x10::lang::VoidFun_0_0::itable<simpleLoop__closure__1> _itable;
    static ::x10aux::itable_entry _itables[2];
    
    virtual ::x10aux::itable_entry* _getITables() { return _itables; }
    
    // closure body
    void __apply() {
        try {
            
            //#line 12 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
            ::x10::lang::Rail< ::chunk* >* sample__90 = ::x10::lang::Rail< ::chunk* >::_make(((x10_long)2ll));
            
        }
        catch (::x10::lang::CheckedThrowable* __exc1) {
            if (::x10aux::instanceof< ::x10::lang::Error*>(__exc1)) {
                ::x10::lang::Error* __lowerer__var__0__ = static_cast< ::x10::lang::Error*>(__exc1);
                {
                    ::x10aux::throwException(::x10aux::nullCheck(__lowerer__var__0__));
                }
            } else
            if (true) {
                ::x10::lang::CheckedThrowable* __lowerer__var__1__ = static_cast< ::x10::lang::CheckedThrowable*>(__exc1);
                {
                    ::x10aux::throwException(::x10aux::nullCheck(::x10::lang::Exception::ensureException(
                                                                   __lowerer__var__1__)));
                }
            } else
            throw;
        }
    }
    
    // captured environment
    
    ::x10aux::serialization_id_t _get_serialization_id() {
        return _serialization_id;
    }
    
    ::x10aux::serialization_id_t _get_network_id() {
        return _network_id;
    }
    
    void _serialize_body(::x10aux::serialization_buffer &buf) {
        
    }
    
    static x10::lang::Reference* _deserialize(::x10aux::deserialization_buffer &buf) {
        simpleLoop__closure__1* storage = ::x10aux::alloc_z<simpleLoop__closure__1>();
        buf.record_reference(storage);
        simpleLoop__closure__1* this_ = new (storage) simpleLoop__closure__1();
        return this_;
    }
    
    simpleLoop__closure__1() { }
    
    static const ::x10aux::serialization_id_t _serialization_id;
    
    static const ::x10aux::serialization_id_t _network_id;
    
    static const ::x10aux::RuntimeType* getRTT() { return ::x10aux::getRTT< ::x10::lang::VoidFun_0_0>(); }
    virtual const ::x10aux::RuntimeType *_type() const { return ::x10aux::getRTT< ::x10::lang::VoidFun_0_0>(); }
    
    const char* toNativeString() {
        return "/home/induk/workspace/testprgms/src/simpleLoop.x10:11-13";
    }

};

#endif // SIMPLELOOP__CLOSURE__1_CLOSURE

//#line 5 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
void simpleLoop::main(::x10::lang::Rail< ::x10::lang::String* >* args) {
    
    //#line 8 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
    ::x10::io::Console::FMGL(OUT__get)()->x10::io::Printer::println(reinterpret_cast< ::x10::lang::Any*>((__extension__ ({ static ::x10::lang::String* strLit__110 = ::x10aux::makeStringLit("hi"); strLit__110; }))));
    {
        
        //#line 9 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
        ::x10::lang::Runtime::ensureNotInAtomic();
        ::x10::lang::FinishState* x10____var0 = ::x10::lang::Runtime::startFinish();
        {
            ::x10::lang::CheckedThrowable* throwable__97 = ::x10aux::class_cast_unchecked< ::x10::lang::CheckedThrowable*>(reinterpret_cast< ::x10::lang::NullType*>(X10_NULL));
            try {
                try {
                    {
                        x10_long nproc__86 = ((x10_long)4ll);
                        ::x10::lang::Rail< x10_boolean >* done__87 = ::x10::lang::Rail< x10_boolean >::_make(((x10_long)8ll));
                        {
                            x10_long proc__88;
                            for (proc__88 = ((x10_long)0ll); ((proc__88) < (nproc__86));
                                 proc__88 = ((proc__88) + (((x10_long)1ll))))
                            {
                                x10_long tproc__89 = proc__88;
                                
                                //#line 10 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
                                ::x10::lang::Runtime::runAsync(reinterpret_cast< ::x10::lang::VoidFun_0_0*>((new (::x10aux::alloc< ::x10::lang::VoidFun_0_0>(sizeof(simpleLoop__closure__1)))simpleLoop__closure__1())));
                            }
                        }
                        
                    }
                }
                catch (::x10::lang::CheckedThrowable* __exc2) {
                    if (true) {
                        ::x10::lang::CheckedThrowable* __lowerer__var__0__ =
                          static_cast< ::x10::lang::CheckedThrowable*>(__exc2);
                        {
                            ::x10::lang::Runtime::pushException(__lowerer__var__0__);
                            ::x10aux::throwException(::x10aux::nullCheck(::x10::lang::Exception::_make()));
                        }
                    } else
                    throw;
                }
                ::x10::compiler::Finalization::plausibleThrow();
            }
            catch (::x10::lang::CheckedThrowable* __exc3) {
                if (true) {
                    ::x10::lang::CheckedThrowable* formal__98 = static_cast< ::x10::lang::CheckedThrowable*>(__exc3);
                    {
                        throwable__97 = formal__98;
                    }
                } else
                throw;
            }
            if ((!::x10aux::struct_equals(reinterpret_cast< ::x10::lang::CheckedThrowable*>(X10_NULL),
                                          throwable__97)))
            {
                if (::x10aux::instanceof< ::x10::compiler::Abort*>(throwable__97))
                {
                    ::x10aux::throwException(::x10aux::nullCheck(throwable__97));
                }
                
            }
            if (true) {
                ::x10::lang::Runtime::stopFinish(x10____var0);
            }
            if ((!::x10aux::struct_equals(reinterpret_cast< ::x10::lang::CheckedThrowable*>(X10_NULL),
                                          throwable__97)))
            {
                if (!(::x10aux::instanceof< ::x10::compiler::Finalization*>(throwable__97)))
                {
                    ::x10aux::throwException(::x10aux::nullCheck(throwable__97));
                }
                
            }
            
        }
    }
}

//#line 4 "/home/induk/workspace/testprgms/src/simpleLoop.x10"
::simpleLoop* simpleLoop::simpleLoop____this__simpleLoop(
  ) {
    return this;
    
}
void simpleLoop::_constructor() {
    this->simpleLoop::__fieldInitializers_simpleLoop();
}
::simpleLoop* simpleLoop::_make() {
    ::simpleLoop* this_ = new (::x10aux::alloc_z< ::simpleLoop>()) ::simpleLoop();
    this_->_constructor();
    return this_;
}


void simpleLoop::__fieldInitializers_simpleLoop() {
 
}
const ::x10aux::serialization_id_t simpleLoop::_serialization_id = 
    ::x10aux::DeserializationDispatcher::addDeserializer(::simpleLoop::_deserializer);

void simpleLoop::_serialize_body(::x10aux::serialization_buffer& buf) {
    
}

::x10::lang::Reference* ::simpleLoop::_deserializer(::x10aux::deserialization_buffer& buf) {
    ::simpleLoop* this_ = new (::x10aux::alloc_z< ::simpleLoop>()) ::simpleLoop();
    buf.record_reference(this_);
    this_->_deserialize_body(buf);
    return this_;
}

void simpleLoop::_deserialize_body(::x10aux::deserialization_buffer& buf) {
    
}

::x10aux::RuntimeType simpleLoop::rtt;
void simpleLoop::_initRTT() {
    if (rtt.initStageOne(&rtt)) return;
    const ::x10aux::RuntimeType** parents = NULL; 
    rtt.initStageTwo("simpleLoop",::x10aux::RuntimeType::class_kind, 0, parents, 0, NULL, NULL);
}

::x10::lang::VoidFun_0_0::itable<simpleLoop__closure__1>simpleLoop__closure__1::_itable(&::x10::lang::Reference::equals, &::x10::lang::Closure::hashCode, &simpleLoop__closure__1::__apply, &simpleLoop__closure__1::toString, &::x10::lang::Closure::typeName);
::x10aux::itable_entry simpleLoop__closure__1::_itables[2] = {::x10aux::itable_entry(&::x10aux::getRTT< ::x10::lang::VoidFun_0_0>, &simpleLoop__closure__1::_itable),::x10aux::itable_entry(NULL, NULL)};

const ::x10aux::serialization_id_t simpleLoop__closure__1::_serialization_id = 
    ::x10aux::DeserializationDispatcher::addDeserializer(simpleLoop__closure__1::_deserialize);
const ::x10aux::serialization_id_t simpleLoop__closure__1::_network_id = 
    ::x10aux::NetworkDispatcher::addNetworkDeserializer(simpleLoop__closure__1::_deserialize,::x10aux::CLOSURE_KIND_ASYNC_CLOSURE);

/* END of simpleLoop */
/*************************************************/
